apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BaseStressTest.cs

```diff
@@ -0,0 +1,102 @@
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Google.Cloud.Spanner.V1;
+using Google.Cloud.Spanner.V1.Logging;
+
+namespace Google.Cloud.Spanner.Data.IntegrationTests
+{
+    public abstract class BaseStressTest
+    {
+        protected const int TargetQps = 100;
+        protected static readonly TimeSpan TestDuration = TimeSpan.FromSeconds(60);
+
+        protected abstract Task<TimeSpan> TestWrite1(Stopwatch sw);
+
+        private async Task<TimeSpan> TestWrite(Stopwatch sw)
+        {
+            await Task.Yield(); //We immediately yield to allow the spawning thread to continue.
+            return await TestWrite1(sw);
+        }
+
+
+        private async Task Timeout(TimeSpan testTime)
+        {
+            await Task.Delay(testTime.Add(TimeSpan.FromSeconds(10)));
+        }
+
+        /// <summary>
+        /// These perf statistics are gathered when LogPerformanceTraces = true.
+        /// Important statistics here are:
+        ///  SpannerTransaction.Commit.Duration: The duration to call spanner server for a commit transaction.
+        ///  Transaction.Begin.Duration: The duration to call spanner to begin a transaction.  This value + commit duration should be greater than measured latency
+        ///    if our transaction prewarming is working properly.
+        ///  Session.CreateTime: The duration to call spanner server for a new session.  This should mostly be done at
+        ///    prewarm.  If "Recordings" is more than .25 * TargetQps, then it means additional sessions had to be allocated and the server probably
+        ///    started falling behind the target Qps.
+        ///  SpannerClient.RawCreateCount: You are looking at "recordings" here to ensure that only `MaximumGrpcChannels` are created.
+        ///  Transaction.CacheHit: How many hits we got on prewarmed transactions.
+        /// </summary>
+        protected async Task<double> TestWriteLatencyWithQps(double queriesPerSecond, TimeSpan testTime)
+        {
+            var sw = Stopwatch.StartNew();
+            var all = new List<Task<TimeSpan>>();
+            var timeout = Timeout(testTime);
+
+            while (sw.Elapsed < testTime)
+            {
+                if (sw.Elapsed.TotalSeconds * queriesPerSecond > all.Count)
+                {
+                    all.Add(TestWrite(Stopwatch.StartNew()));
+                }
+            }
+
+            var completedTask = await Task.WhenAny(Task.WhenAll(all), timeout);
```

Comment:
pretty much.  at some earlier point before I had the thread never yield, it might have been more useful.
in the current iteration, you are right -- I can just add the extra 10 seconds at the end.